src/include/libpmemset.h

```diff
@@ -0,0 +1,268 @@
+		const wchar_t *file);
+
+int pmemset_source_from_file_paramsW(struct pmemset_source **src,
+		const wchar_t *file);
+
+int pmemset_source_from_temporaryU(struct pmemset_source **src,
+		const char *dir);
+
+int pmemset_source_from_temporaryW(struct pmemset_source **src,
+		const wchar_t *dir);
+#endif
+
+int pmemset_source_fallocate(struct pmemset_source *src, int flags,
+		size_t offset, size_t size);
+
+int pmemset_source_delete(struct pmemset_source **src);
+
+/* part setup */
+
+struct pmemset_part;
+struct pmemset_part_descriptor;
+struct pmemset_part_map;
+struct pmemset_part_shutdown_state_data;
+
+enum pmemset_part_state {
+	/*
+	 * The pool state cannot be determined because of errors during
+	 * retrieval of device information.
+	 */
+	PMEMSET_PART_STATE_INDETERMINATE,
+
+	/*
+	 * The pool is internally consistent and was closed cleanly.
+	 * Application can assume that no custom recovery is needed.
+	 */
+	PMEMSET_PART_STATE_OK,
+
+	/*
+	 * The pool is internally consistent, but it was not closed cleanly.
+	 * Application must perform consistency checking and custom recovery
+	 * on user data.
+	 */
+	PMEMSET_PART_STATE_OK_BUT_INTERRUPTED,
+
+	/*
+	 * The pool can contain invalid data as a result of hardware failure.
+	 * Reading the pool is unsafe.
+	 */
+	PMEMSET_PART_STATE_CORRUPTED,
+};
+
+int pmemset_part_descriptor_new(struct pmemset_part_descriptor **part,
+		struct pmemset *set, struct pmemset_source *src, size_t offset,
+		size_t length);
+
+int pmemset_part_descriptor_pread_mcsafe(struct pmemset_part_descriptor *part,
+		void *dst, size_t size, size_t offset);
+
+int pmemset_part_descriptor_pwrite_mcsafe(struct pmemset_part_descriptor *part,
+		void *dst, size_t size, size_t offset);
+
+int pmemset_part_finalize(struct pmemset *set,
+		struct pmemset_part_descriptor *descriptor,
+		const struct pmemset_header *header_in,
+		struct pmemset_header *header_out,
+		const struct pmemset_part_shutdown_state_data *sds_in,
+		struct pmemset_part_shutdown_state_data *sds_out,
+		enum pmemset_part_state *state);
+
+int pmemset_part_address(struct pmemset_part *part);
+
+int pmemset_part_length(struct pmemset_part *part);
+
+int pmemset_part_first(struct pmemset *set, struct pmemset_part **part);
+
+int pmemset_part_next(struct pmemset *set, struct pmemset_part **part);
+
+int pmemset_part_by_address(struct pmemset *set, struct pmemset_part **part,
+		void *addr);
+
+int pmemset_part_drop(struct pmemset_part *part);
+
+int pmemset_part_map_new(struct pmemset_part_map **pmap,
+		struct pmemset_part **part,
+		const struct pmemset_header *header_in,
+		struct pmemset_header *header_out,
+		const struct pmemset_part_shutdown_state_data *sds_in,
+		struct pmemset_part_shutdown_state_data *sds_out,
+		enum pmemset_part_state *state);
+
+int pmemset_part_map_drop(struct pmemset_part_map **pmap);
+
+int pmemset_part_map_address(struct pmemset_part_map *pmap);
+
+int pmemset_part_map_length(struct pmemset_part_map *pmap);
```

Comment:
this function should look different (merge in one function which returns structure with addr and len)